### PR TITLE
ENH: draft take/searchsorted

### PR DESCRIPTION
**Pull request by mdhaber**
I think this is the gist of it, but might need to work on edge cases, and I need to get the test working.

### CODE REVIEW

File: marray/tests/test_marray.py [hunk 1]
```diff
@@ -413,6 +413,49 @@ def test_statistical_array(f_name, keepdims, xp=np, dtype='float64', seed=None):
...
+    if sorted:
+        x = mxp.sort(x)
+
+    x = mxp.asarray(x, mask=x_mask)
+    y = mxp.asarray(y, mask=y_mask)
+
+    kwargs = {'side': side} if sorted else {'side': side, 'sorter': mxp.argsort(x)}
+    out = mxp.searchsorted(x, y, **kwargs)
+
+    for j in out.size:
```
Comment (keewis) [inline marray/tests/test_marray.py:438]:
```suggestion
    for j in range(out.size):
```

File: marray/__init__.py [hunk 2]
```diff
@@ -253,8 +253,10 @@ def fun(*args, name=name, **kwargs):
         setattr(mod, name, fun)
 
     ## Indexing Functions
-    # To be written:
-    # take
+    def take(x, indices, /, *, axis=None):
+        data = xp.take(x.data, indices.data, axis=axis)
+        mask = xp.take(x.mask, indices.data, axis=axis) + indices.mask
```
[the remark this answers is not in the document]
Comment (mdhaber) [reply marray/__init__.py:258]: What I've been doing is to check whether the input has a mask attribute: if so, use it, otherwise, use the whole object.
